### PR TITLE
Install OpenSSL in Windows pipeline

### PR DESCRIPTION
**Pull request by svix-andor**
Let's see if this is enough to get Windows CI working again

### CODE REVIEW

File: .github/workflows/server-release.yml [hunk 1]
```diff
@@ -42,6 +42,10 @@ jobs:
         if: ${{ matrix.target == 'x86_64-unknown-linux-musl' }}
         run: sudo apt-get install -y musl-dev musl-tools
 
+      - name: Install OpenSSL for Windows
+        if: ${{ matrix.target == 'x86_64-pc-windows-gnu' }}
```
Comment (tasn) [inline .github/workflows/server-release.yml:46]: Are we using windows-gnu?

Comment (svix-andor) [reply .github/workflows/server-release.yml:46]: https://github.com/svix/svix-webhooks/blob/main/.github/workflows/server-release.yml#L15
yep

File: .github/workflows/server-release.yml [hunk 2]
```diff
@@ -1,8 +1,9 @@
 name: Server Release
 
 on:
-  release:
-    types: [published]
+  push:
+    branches:
+      - main
```
Comment (tasn) [inline .github/workflows/server-release.yml:6]: 🤔

Comment (svix-andor) [reply .github/workflows/server-release.yml:6]: Accidentally pushed this here while debugging this on my forked copy.

Comment (tasn) [reply .github/workflows/server-release.yml:6]: Keep this comment open, so that we remember to fix it before merging.

**Pull request closed without merging**